### ISSUE TITLE
DNS-01 Digitalocean 404 - HTTP 404: not_found: The resource you were accessing could not be found

### ISSUE THREAD

**Issue by Temikus** [go-acme/lego#1066]
I was having issues setting up wildcard certificates for my services via traefik. Digging deeper into the problem it appears that the issue lies somewhere in LEGO, as I can reproduce it via CLI but I can obtain certs via `certbot` without issues.

Repro (note: the issue is reproducible in the same way on prod letsencrypt servers, I'm just using staging so I don't exhaust my tokens):
```
docker run -it --rm --name lego \
            -e "DO_AUTH_TOKEN=XXXX" \
            goacme/lego \
            --email="acme@temik.me" --domains="mywebsite.dev" --dns="digitalocean" \
            --server="https://acme-staging-v02.api.letsencrypt.org/directory" \
            run

...

2020/02/23 05:27:39 [INFO] [mywebsite.dev] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/40169521
2020/02/23 05:27:39 [INFO] [mywebsite.dev] acme: Could not find solver for: tls-alpn-01
2020/02/23 05:27:39 [INFO] [mywebsite.dev] acme: Could not find solver for: http-01
2020/02/23 05:27:39 [INFO] [mywebsite.dev] acme: use dns-01 solver
2020/02/23 05:27:39 [INFO] [mywebsite.dev] acme: Preparing to solve DNS-01
2020/02/23 05:27:40 [INFO] [mywebsite.dev] acme: Cleaning DNS-01 challenge
2020/02/23 05:27:40 [WARN] [mywebsite.dev] acme: error cleaning up: digitalocean: unknown record ID for '_acme-challenge.mywebsite.dev.'
2020/02/23 05:27:40 [INFO] Deactivating auth: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/40169521
2020/02/23 05:27:41 Could not obtain certificates:
	acme: Error -> One or more domains had a problem:
[mywebsite.dev] [mywebsite.dev] acme: error presenting token: digitalocean: HTTP 404: not_found: The resource you were accessing could not be found.
```

Certbot performs without issues via:
```
docker run -it --rm --name certbot \
            -v "/etc/letsencrypt:/etc/letsencrypt" \
            -v "/var/lib/letsencrypt:/var/lib/letsencrypt" \
            certbot/dns-digitalocean certonly \
            --server https://acme-staging-v02.api.letsencrypt.org/directory \
            --dns-digitalocean \
            --dns-digitalocean-credentials /etc/letsencrypt/do.ini \
            -d *.mywebsite.dev -d mywebsite.dev
```

If I manually access the API, I don't get a 404 either:

```
λ  export TOKEN=XXXXXX                                                 

λ curl -X GET "https://api.digitalocean.com/v2/domains/mywebsite.dev/records" -H "Authorization: Bearer $TOKEN"
{"domain_records":[{"id":90745039,"type":"SOA","name":"@","data":"1800","priority":null,"port":null,"ttl":1800,"weight":null,"flags":null,"tag":null},{"id":90745042,"type":"NS","name":"@","data":"ns1.digitalocean.com","priority":null,"port":null,"ttl":1800,"weight":null,"flags":null,"tag":null},{"id":90745043,"type":"NS","name":"@","data":"ns2.digitalocean.com","priority":null,"port":null,"ttl":1800,"weight":null,"flags":null,"tag":null},{"id":90745044,"type":"NS","name":"@","data":"ns3.digitalocean.com","priority":null,"port":null,"ttl":1800,"weight":null,"flags":null,"tag":null}],"links":{},"meta":{"total":4}}%
```

I'm honestly stumped here. I'm willing to provide any info you might require, any debug logs, etc.

EDIT: Added some details.

**Comment by Temikus** [comment]
Ok, so the issue was with this specific domain. Looks like NS records took 48+ hours to propagate.

Should I leave this bug to see if error messaging can be improved or close it out?

**Comment by Temikus** [comment]
I've since ran into this issue again and want to leave a breadcrumb for other folks that run into this, since the other bug report is less discoverable.

The issue was the same as: https://github.com/go-acme/lego/issues/1147

Situation on my end was:
- I use split DNS (i.e. have the domain records in question defined locally) 
- Forgot that I use DNS hijacking for my pihole installation, which caused the traefik to be pointed to internal record, regardless of resolvers set and not find the TXT and SOA records.

After excluding the server in question from DNS hijacking I was immediately able to make cert requests. Seems that `Lego` checks the records via DNS on its' own and `certbot` doesn't, hence why I was able to get a cert through `certbot`
